modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx

```diff
@@ -588,11 +603,16 @@ export default function ActionBuilder({
 									aria-label={Liferay.Language.get(
 										'choose-an-object'
 									)}
-									error={errors.objectDefinitionId}
+									error={
+										errors.objectDefinitionExternalReferenceCode
+									}
 									onChange={handleSelectObject}
 									options={objectsOptions}
 									value={
-										values.parameters?.objectDefinitionId
+										values.parameters
+											?.objectDefinitionExternalReferenceCode
+											? `${values.parameters?.objectDefinitionExternalReferenceCode},${values.parameters?.objectDefinitionId}`
```

Comment:
@danvb112 I think you can remove the optional chaining on `${values.parameters?.objectDefinitionExternalReferenceCode}...`